src/core/src/opsets/opset.cpp

```diff
@@ -6,31 +6,77 @@
 
 ngraph::OpSet::OpSet(const ov::OpSet& opset) : ov::OpSet(opset) {}
 
+const ngraph::FactoryRegistry<ngraph::Node>& ngraph::OpSet::get_factory_registry() {
+    NGRAPH_SUPPRESS_DEPRECATED_START
+    for (const auto& builder : m_factory->get_builders()) {
+        m_factory_registry.register_factory(builder.first, builder.second);
+    }
+    return m_factory_registry;
+    NGRAPH_SUPPRESS_DEPRECATED_END
+}
+
+ov::OpSet::OpSet() {
+    m_factory = std::unique_ptr<ov::opset_factory::Factory>(new ov::opset_factory::Factory{"default"});
+}
+
+ov::OpSet::OpSet(const std::string& name) {
+    m_factory = std::unique_ptr<ov::opset_factory::Factory>(new ov::opset_factory::Factory{name});
+}
+
+ov::OpSet::OpSet(const ov::OpSet& opset) {
+    m_op_types = opset.m_op_types;
+    m_name_type_info_map = opset.m_name_type_info_map;
+    m_case_insensitive_type_info_map = opset.m_case_insensitive_type_info_map;
+    m_factory = std::unique_ptr<ov::opset_factory::Factory>(new ov::opset_factory::Factory(*opset.m_factory));
+}
+
+ov::OpSet ov::OpSet::operator=(const ov::OpSet& opset) {
+    m_op_types = opset.m_op_types;
+    m_name_type_info_map = opset.m_name_type_info_map;
+    m_case_insensitive_type_info_map = opset.m_case_insensitive_type_info_map;
+    m_factory = std::unique_ptr<ov::opset_factory::Factory>(new ov::opset_factory::Factory(*opset.m_factory));
+    return *this;
+}
+
+ov::OpSet::~OpSet() {}
+
 std::mutex& ov::OpSet::get_mutex() {
     static std::mutex opset_mutex;
     return opset_mutex;
 }
 
+void ov::OpSet::insert(const std::string& name,
+                       const ov::NodeTypeInfo& type_info,
+                       const std::function<ov::Node*()> builder) {
+    std::lock_guard<std::mutex> guard(get_mutex());
+
+    m_op_types.insert(type_info);
+    m_name_type_info_map[name] = type_info;
+    m_case_insensitive_type_info_map[to_upper_name(name)] = type_info;
+    m_factory->register_type(type_info, std::move(builder));
+}
+
 ov::Node* ov::OpSet::create(const std::string& name) const {
     auto type_info_it = m_name_type_info_map.find(name);
     if (type_info_it == m_name_type_info_map.end()) {
         NGRAPH_WARN << "Couldn't create operator of type: " << name << " . Operation not registered in opset.";
         return nullptr;
     }
-    return m_factory_registry.create(type_info_it->second);
+    return m_factory->create(type_info_it->second);
 }
 
 ov::Node* ov::OpSet::create_insensitive(const std::string& name) const {
     auto type_info_it = m_case_insensitive_type_info_map.find(to_upper_name(name));
-    return type_info_it == m_case_insensitive_type_info_map.end() ? nullptr
-                                                                  : m_factory_registry.create(type_info_it->second);
+    return type_info_it == m_case_insensitive_type_info_map.end() ? nullptr : m_factory->create(type_info_it->second);
 }
 
 const ov::OpSet& ov::get_opset1() {
-    static OpSet opset;
+    static OpSet opset("get_opset1");
```

Comment:
```suggestion
    static OpSet opset("opset1");
```

Reply:
> Could you please check how load time is changed?

Load time isn't changed, please, find time tests results in a ticket